guides/source/upgrading_ruby_on_rails.md

```diff
@@ -257,6 +257,26 @@ See the [Testing Rails Applications](https://guides.rubyonrails.org/testing.html
 
 If you run a single file's tests (`bin/rails test test/models/user_test.rb`), `test:prepare` will not run before it.
 
+### Import syntax from `@rails/ujs` is modified
+
+Starting from Rails 7.1, the syntax for importing modules from `@rails/ujs` is modified. Rails no longer supports the
+direct import of a module from `@rails/ujs`. If users create and build a JavaScript file with the line `import
+{ fileInputSelector } from "@rails/ujs"`, they will encounter the error `export 'fileInputSelector'
+(imported as 'fileInputSelector') was not found in '@rails/ujs' (possible exports: default)`.
```

Comment:
```suggestion
direct import of a module from `@rails/ujs`.

For example, attempting to import a function from the library will fail:

\```
import { fileInputSelector } from "@rails/ujs"
#=> export 'fileInputSelector' (imported as 'fileInputSelector') was not found in '@rails/ujs' (possible exports: default)
\```
```

The code fences in the suggested patch might be weird, but the point is I think creating a separate code block for the failure case (with the resulting error on a single line) may help improve readability and anyone searching for this error explicitly.